Add documentation for Directus UI Components (#20675)
directus/directus: docs/extensions/leveraging-ui-components.md

---

> description: Learn how to use the Directus UI components in your extension projects.

Reviewer: ```suggestion
description: Learn how to use the Directus UI components in your app extensions.
```

Removing 'projects' as that has a defined meaning within Directus

contributors: Esther Agbaje
---

> # Leveraging Directus UI Components

Reviewer: ```suggestion
# Using Directus UI Components
```

> Directus provides a set of reusable UI components that can be used in when building extensions. These components are
> designed to provide a consistent user experience across different projects and extensions.

Reviewer: ```suggestion
Directus provides a set of reusable UI components that can be used in when building extensions. These components are used by Directus itself, and are
designed to provide a consistent user experience across different parts of the Directus Data Studio.
```

## Components Playground

> Explore the available components on the [Histoire Components Playground](https://components.directus.io/). Here, you can
> interact with different components and experiment with variations before implementing them in your project.

Reviewer: ```suggestion
Explore the available components in our [Directus Components Playground](https://components.directus.io/). Here, you can
see all available components and experiment with variations before implementing them in your extensions.
```

Reviewer: I don't think we need to expose the readers to this new term and tool - Histoire.

> ![Directus UI Components](https://marketing.directus.app/assets/46d72f1a-5d9f-49f8-a09a-b2ffc5200812.gif)

Reviewer: This alt text needs to basically be a textual description of the important features of the playground shown in the gif.

> ## Usage within Extensions

Reviewer: ```suggestion
## Extension Usage
```

> Directus UI components are globally registered, making them accessible throughout your extension project without the
> need for explicit imports. Here’s a basic example:
> 
> ```tsx
> <VButton>
>   My Button
> </VButton>
> ```

Reviewer: ```suggestion
Directus UI components are globally registered, making them accessible throughout your extension project without the
need for explicit imports. Here’s an example:
```

Reviewer: A reminder: https://justsimply.dev/

Reviewer 2: Was this supposed to remove the example?

## Customizing Styles

Each component exports CSS custom properties (variables) that can be targeted for style overrides.

For example, to adjust the text color of a button on hover:

```tsx
// give the button a class
<VButton class="my-button">
  My Button
</VButton>

// customize the style
.my-button {
  --v-button-color-hover: black;
}
```

> ::: tip Explore CSS Variables

Reviewer: ```suggestion
::: tip Explore Component CSS Variables
```

Refer to the full list of component based CSS variables in our codebase
[in our codebase](https://github.com/directus/directus/tree/main/app/src/components).

:::

## Creating a Custom UI Component

> The Directus UI components are designed with flexibility and customization in mind, enabling you to build your custom UI
> components.
> 
> We exposes a couple of CSS variables for both light and dark themes. These variables are useful when building a custom
> component and want it to look native to Directus.
> 
> Here are some examples of CSS variables: `--border-normal`, `--foreground-normal` `-purple`, `--module-background`, and
> `--overlay-color`.
> 
> ::: tip Explore Light and Dark Theme CSS Variables
> 
> Refer to our [codebase](https://github.com/directus/directus/tree/main/app/src/styles/themes) for a full list of CSS
> variables.
> 
> :::

Reviewer: ```suggestion
The Directus UI components are designed with flexibility and customization in mind. However, you may need to create your own components using shared styling. Directus exposes several CSS variables for both light and dark themes.

Examples of CSS variables include `--border-normal`, `--foreground-normal` `-purple`, `--module-background`, and
`--overlay-color`.

::: tip Explore Light and Dark Theme CSS Variables

Refer to our [codebase](https://github.com/directus/directus/tree/main/app/src/styles/themes) for a full list of CSS
variables.

:::
```
